app/src/main/java/com/codesoom/assignment/TaskController.java

```diff
@@ -0,0 +1,155 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.enums.HttpMethodType;
+import com.codesoom.assignment.error.ClientError;
+import com.codesoom.assignment.models.Path;
+import com.codesoom.assignment.models.RequestTaskDTO;
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.service.TaskService;
+import com.codesoom.assignment.utils.JsonUtil;
+import com.codesoom.assignment.utils.StringUtil;
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Optional;
+
+public class TaskController {
+    private static final TaskController taskController = new TaskController();
+    private final TaskService taskService = TaskService.getInstance();
+
+    private TaskController() {
+    }
+
+    public static TaskController getInstance() {
+        return taskController;
+    }
+
+    public void requestMapping(HttpExchange exchange,
+                               HttpMethodType method,
+                               Path path,
+                               String requestBody) throws IOException {
+        if (isRequiredPathVariable(method) && path.getPathVariable() == null) {
+            ClientError.badRequest(exchange);
+            return;
+        }
+
+        if (isRequiredRequestBody(method) && requestBody.isBlank()) {
+            ClientError.badRequest(exchange);
+            return;
+        }
+
+        switch (method) {
+            case GET:
+                if (path.getPathVariable() == null) {
+                    gets(exchange);
+                    return;
+                }
+
+                if (!StringUtil.isNumeric(path.getPathVariable())) {
```

Comment:
이런 모양이 되도록 수정해 보세요.

```suggestion
                if (!path.hasTaskId()) {
```

이렇게 할 수도 있겠죠.

```suggestion
                if (!path.isRepresentsTaskIdentity()) {
```

Reply:
Path 객체에게 numeric 타입을 검증하는 메서드를 생성해주며, path 검증은 Path 객체에만 의존하도록 변경하였습니다!
```java
    if (!path.isPathVariableNumeric()) {
        ClientError.methodArgumentTypeMismatch(exchange);
        return;
    }


// Path 객체 내부
    public boolean isPathVariableNumeric() {
        return this.pathVariable.chars()
                .allMatch(Character::isDigit);
    }
```